From be769d634012e5551272ff58e53f7125d6a4d9cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 31 Jul 2025 23:54:19 +0200
Subject: [PATCH] Add D-Bus service file

And mark the application as D-Bus activatable. This allows application
launchers to activate it via D-Bus.
---
 data/com.rafaelmardojai.Blanket.desktop.in |  1 +
 data/com.rafaelmardojai.Blanket.service.in |  3 +++
 data/meson.build                           | 10 ++++++++++
 3 files changed, 14 insertions(+)
 create mode 100644 data/com.rafaelmardojai.Blanket.service.in

diff --git a/data/com.rafaelmardojai.Blanket.desktop.in b/data/com.rafaelmardojai.Blanket.desktop.in
index 806c13be..6cbaba60 100644
--- a/data/com.rafaelmardojai.Blanket.desktop.in
+++ b/data/com.rafaelmardojai.Blanket.desktop.in
@@ -9,5 +9,6 @@ Type=Application
 Keywords=Concentrate;Focus;Noise;Productivity;Sleep;
 Categories=AudioVideo;Audio;GTK;
 StartupNotify=true
+DBusActivatable=true
 # Translators: Do NOT translate or transliterate this text (these are enum types)!
 X-Purism-FormFactor=Workstation;Mobile;
diff --git a/data/com.rafaelmardojai.Blanket.service.in b/data/com.rafaelmardojai.Blanket.service.in
new file mode 100644
index 00000000..7c0bd538
--- /dev/null
+++ b/data/com.rafaelmardojai.Blanket.service.in
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=com.rafaelmardojai.Blanket
+Exec=@bindir@/blanket --gapplication-service
diff --git a/data/meson.build b/data/meson.build
index c067cdac..d5674042 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -52,3 +52,13 @@ install_data(
   'com.rafaelmardojai.Blanket-symbolic.svg',
   install_dir: join_paths(get_option('datadir'), 'icons/hicolor/symbolic/apps')
 )
+
+service_conf = configuration_data()
+service_conf.set('bindir', join_paths(get_option('prefix'), get_option('bindir')))
+configure_file(
+  input: 'com.rafaelmardojai.Blanket.service.in',
+  output: 'com.rafaelmardojai.Blanket.service',
+  configuration: service_conf,
+  install: true,
+  install_dir: join_paths(get_option('datadir'), 'dbus-1', 'services')
+)
